smart_open/http.py

```diff
@@ -48,6 +48,7 @@ def __init__(self, url, mode='r', buffer_size=DEFAULT_BUFFER_SIZE,
         self.buffer_size = buffer_size
         self.mode = mode
 
+        _HEADERS.update(headers)
```

Comment:
Mutating global vars is a no-no (make a copy first).

Reply:
Ok, I'll change it.

Reply:
Better?

Reply:
I think the other way makes more sense: overwriting (a copy of) our defaults with the user-supplied headers. Two reasons:

1. So that user-supplied headers takes precedence over defaults.
2. So that we don't mutate the user-supplied `headers` dict. It could lead to nasty surprises, even now -- since the default `headers` param is mutable, changing it will affect *all* future invocation of `__init__`, not just this one call!

The standard Python pattern is to avoid mutable params, using `headers=None` instead to denote the default, which is then set in the function body.

Reply:
Makes sense.